lib/commons/dom/get-visible-child-text-rects.js

```diff
@@ -0,0 +1,80 @@
+import { getNodeFromTree, memoize } from '../../core/utils';
+import { sanitize } from '../text';
+import { getRectCenter, isPointInRect, getIntersectionRect } from '../math';
+import getOverflowHiddenAncestors from './get-overflow-hidden-ancestors';
+
+/**
+ * Get the visible text client rects of a node.
+ * @method getVisibleChildTextRects
+ * @memberof axe.commons.dom
+ * @instance
+ * @param {Element} node
+ */
+const getVisibleChildTextRects = memoize(
+  function getVisibleChildTextRectsMemoized(node) {
+    const vNode = getNodeFromTree(node);
+    const nodeRect = vNode.boundingClientRect;
+    const clientRects = [];
+    const overflowHiddenNodes = getOverflowHiddenAncestors(vNode);
+
+    Array.from(node.childNodes).forEach(elm => {
+      if (elm.nodeType !== 3 || sanitize(elm.textContent) === '') {
+        return;
+      }
+
+      const contentRects = getContentRects(elm);
+      if (isOutsideNodeBounds(contentRects, nodeRect)) {
+        return;
+      }
+
+      contentRects.forEach(contentRect => {
```

Comment:
I keep tripping up in this part of the function trying to work out what this does. I think moving this forEach into a new function would help a lot with that.

```suggestion
      const visibleRects = removeHiddenRects(contentRects, overflowHiddenNodes)
      clientRects.push(...visibleRects);
```